src/com/xmslyz/luhn/main/Main.java

```diff
@@ -0,0 +1,14 @@
+package com.xmslyz.luhn.main;
+
+import com.xmslyz.luhn.logic.LuhnAlgorithmEngine;
+
+public class Main {
+    public static void main(String[] args) {
+        LuhnAlgorithmEngine luhnAlgorithmEngine = new LuhnAlgorithmEngine();
```

Comment:
ta linijka powinna wyglądać:
```
  LuhnAlgorithm luhnAlgorithm = new LuhnAlgorithmEngine();
```
Interfejs jest kontraktem, jaki jego implementacje spełniają. Wszedzie gdzie to możliwe, powinieneś starać się używać raczej interfejsu, niż jego konkretnej implementacji.  
Interfejs jest pewną abstrakcją, która pozwala ugenerycznić algorytm. Jeśli dodasz kolejną implementację algorytmu Luhna (np. działającą znacznie szybciej, ale tylko dla małych liczb), to gdy będziesz chciał jej użyć zamiast `LuhnAlgorithmEngine`, będziesz musiał wszystkie miejsca gdzie ona jest wołana pozmieniać. A jeśli wszędzie jest interfejs, to zmienisz tylko jedno miejsce: dokładnie tą linijkę, gdzie tworzony jest `LuhnAlgorithmEngine`.